src/components/PokemonList/PokemonList.tsx

```diff
@@ -1,22 +1,48 @@
-import { useEffect, useState } from "react";
-import { Pokemon } from '@/types/pokemon';
+import { useEffect, useState, useRef } from "react";
+import { PokemonType } from '@/types/pokemonType';
+import { ErrorBox } from '@/components/ErrorBox/ErrorBox';
+import { Loader } from '@/components/Loader/Loader';
 import { fetchAllPokemons } from '@/api/pokemonApi';
-import styles from '@/styles/PokemonList.module.css';
+import styles from '@/components/PokemonList/PokemonList.module.css';
 
 export const PokemonList = () => {
-  const [pokemons, setPokemons] = useState<Pokemon[]>([]);
+  const [pokemons, setPokemons] = useState<PokemonType[]>([]);
+  const isMounted = useRef(false);
+  const [errorArr, setErrArr] = useState<Error[]>([]);
+
+  const fetchIfMounted = async () => {
```

Comment:
```suggestion
  const fetchPokemon= async () => {
```

It's always gonna fetch when called, is it not?

Reply:
It is goanna fetch when called, but not if DISMOUNTED before done fetching. I tried to communicate through the name that this helper function does a bit more validation then just fetch. but I guess it's about conventions. I do also get the need for simplicity, but it may be better with the new terminology - fetchIfNotIgnored(). I also already use fetchAllPokemons() as my Api service method.
Is it better now?